--- Dockerfile	2025-12-08 11:10:27.822473294 +0100
+++ Superdockerfile	2026-03-02 11:10:25.677152717 +0100
@@ -8,6 +8,8 @@
 ##                                                                        ##
 ############################################################################
 
+# A Dockerfile where everything is performed by the `root` user... brrr...
+
 # Creating build image. This image only installs the basic SeaCoral
 # dependencies.
 # This will not provide a useful seacoral instance though, as no useful tool is
@@ -20,6 +22,13 @@
  && sh ./hack-sudoers.sh                                                       \
  && rm hack-sudoers.sh
 
+RUN mkdir -p ~/bin \
+ && exec > ~/bin/sudo \
+ && echo '#!/bin/sh' \
+ && echo 'exec "$@"' \
+ && chmod +x ~/bin/sudo \
+ && echo >>~/.profile 'export PATH="$HOME/bin:$PATH"'
+
 ################################################################################
 
 # Creating the base image. It comes with libfuzzer, klee, and cbmc,
@@ -63,15 +72,6 @@
 # Creating the executable image. It comes with base tools and seacoral itself
 FROM base AS executable
 
-# Create a dedicated user
-RUN adduser --comment "Reef explorer" diver \
- && adduser --group divers \
- && adduser diver divers \
- && usermod --append --groups sudo diver \
- && echo >>/etc/sudoers "diver ALL = NOPASSWD:ALL"
-USER diver
-WORKDIR /home/diver
-
 # Be sure to source `~/.profile` so we don't have to `eval $(opam
 # env)` manually.
 SHELL ["/bin/bash","-l","-c"]
@@ -150,9 +150,9 @@
 # We artificially limit parallelism when running tests as many of the
 # underlying tools are sensitive to CPU and memory availability.
 ARG RUN_TESTS_JOBS=1
-COPY --chown=diver:divers dune* Makefile* seacoral/
-COPY --chown=diver:divers src seacoral/src/
-COPY --chown=diver:divers test seacoral/test/
+COPY                      dune* Makefile* seacoral/
+COPY                      src seacoral/src/
+COPY                      test seacoral/test/
 RUN make -C seacoral apply-frama-c-patch
 RUN if [ "$RUN_TESTS" = yes ];                                                 \
     then time make -C seacoral ci-tests CI_TESTS_JOBS=$RUN_TESTS_JOBS;         \
@@ -169,7 +169,7 @@
 # Still, we remove the seacora/.git directory afterwards to accelerate
 # opam pinning (which appears to copy the whole source tree)
 ARG SKIP_INSTALL=no
-COPY --chown=diver:divers . seacoral
+COPY                      . seacoral
 RUN make -C seacoral reset-versions                                            \
  && git  -C seacoral restore docker                                            \
  && make -C seacoral                                                           \
